utils.py

```diff
@@ -275,25 +273,93 @@ def set(self, lru_cache):
 ip_cache = LRUCacheSingleton()
 
 
-def is_ip_per_task(app):
-    """
-    Return whether the application is using IP-per-task.
-    :param app:  The application to check.
-    :return:  True if using IP per task, False otherwise.
-    """
-    return app.get('ipAddress') is not None
+def get_app_networking_mode(app):
+    mode = 'host'
 
+    if app.get('ipAddress'):
+        mode = 'container'
 
-def is_user_network(app):
-    """
-    Returns True if container network mode is set to USER
-    :param app:  The application to check.
-    :return:  True if using USER network, False otherwise.
-    """
-    c = app.get('container', {})
-    return c is not None and c.get('type', '') == 'DOCKER' and \
-        c.get('docker', {})\
-        .get('network', '') == 'USER'
+    _mode = app.get('container', {})\
+               .get('docker', {})\
+               .get('network', '')
+    if _mode == 'USER':
+        mode = 'container'
+    elif _mode == 'BRIDGE':
+        mode = 'container/bridge'
+
+    networks = app.get('networks', [])
+    for n in networks:
+        # Modes cannot be mixed, so assigning the last mode is fine
+        mode = n.get('mode', 'host')
+
+    return mode
+
+
+def get_task_ip(task, mode):
```

Comment:
That is technically new functionality that I would prefer not to support with this PR. The primary goal here is to allow existing marathon apps to work with marathon-lb after upgrading to marathon 1.5. We can address feature enhancements separately.